collections/_development_standards/platform-integration.md

```diff
@@ -168,6 +168,62 @@ public static class AddDatabaseExtension
 
 As used in: [Courses API AddDatabaseExtension.cs](https://github.com/SkillsFundingAgency/das-courses-api/blob/2c83a21b65bde54ee8ceed9d29c812880cf401bf/src/SFA.DAS.Courses.Api/AppStart/AddDatabaseExtension.cs)
 
+#### Example - Passing a SqlConnection into a DbContext:
+
+Sometimes there may be a need to access data outside of a DbContext e.g. when using Dapper, and additionaly data access may also be required in the same solution inside a DbContext; in which case a shared IDbConnection may be configured outside and then passed into the DbContext.
+
+```csharp
+private readonly IDbConnection _connection;
+
+public AppDataContext()
+{
+}
+
+public AppDataContext(IDbConnection connection, DbContextOptions<AssessorDbContext> options)
+    : base(options)
+{
+    _connection = connection;
+}
+
+protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+{
+    optionsBuilder.UseSqlServer(_connection as DbConnection, options =>
+         options.EnableRetryOnFailure(3));
+}
+
+```
+As used in: [Assessor API AssessorDbContext.cs](https://github.com/SkillsFundingAgency/das-assessor-service/blob/c2587b4055d45024a02ea24d85323980383f2cd7/src/SFA.DAS.AssessorService.Data/AssessorDbContext.cs)
+
+An example of the database extension being added for the AppStart when using StructureMap for DI:
+
+```csharp
+public static class DatabaseExtensions
+{
+    private const string AzureResource = "https://database.windows.net/";
+
+    public static void AddDatabaseRegistration(this ConfigurationExpression config, string environment, string sqlConnectionString)
+    {
+        config.For<IDbConnection>().Use($"Build IDbConnection", c => {
+            var azureServiceTokenProvider = new AzureServiceTokenProvider();
```

Comment:
`AzureServiceTokenProvider` from `Microsoft.Azure.Services.AppAuthentication` is now marked as legacy [and isn't recommended for use](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication#:~:text=Microsoft.Azure.Services,Identity%20Migration%20Guidance.). If we don't want to fix that in the example here it should probably be at least called out as a caveat in the hope someone implements the new package in future.

Reply:
Added a note to call out legacy status of AzureServiceTokenProvider

Reply:
@ChrisJWoodcock Can the note include a link to https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication?

Reply:
Added